src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -649,6 +700,54 @@ class PostgresEntityDataQueryService(
         }
     }
 
+    /**
+     * Updates the last_write column in [IDS] table for the given entities.
+     */
+    private fun updateLastWrite(
+            connection: Connection,
+            entitySetId: UUID,
+            entities: Collection<UUID>,
+            partitionsInfo: PartitionsInfo = partitionManager.getEntitySetPartitionsInfo(entitySetId)
+    ): Int {
+        val idsArr = PostgresArrays.createUuidArray(connection, entities)
+        val partitionsArr = PostgresArrays.createIntArray(
+                connection,
+                entities.map { getPartition(it, partitionsInfo.partitions.toList()) }
```

Comment:
It's probably better to pull the `partitionsInfo.partitions.toList()` out of the `.map` in case there's a very large number of entities since toList() requires iterating over all partitions

Reply:
👍